### PR TITLE
vimPlugins.copilot-lualine: init at 2024-09-03

### PR DESCRIPTION
**Pull request by rsniezek**
## Description of changes
Added vim Plugin for copilot.lua integration with lualine. Works locally.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by rsniezek** [comment]
@figsoda sorry for nagging you but any chance you could take a look at this? Or ping someone with merge rights? Thanks!

**Review by PerchunPak — CHANGES_REQUESTED** [review]
I noticed this plugin relies on `copilot-lua` & `lualine-nvim`. Would it be possible to add them as dependencies, similar to [here](https://github.com/NixOS/nixpkgs/blob/bd4f00ac7fa6b641777b5a7fb08911c456b4ed01/pkgs/applications/editors/vim/plugins/overrides.nix#L197-L199)?

**Comment by rsniezek** [comment]
> I noticed this plugin relies on `copilot-lua` & `lualine-nvim`. Would it be possible to add them as dependencies, similar to [here](https://github.com/NixOS/nixpkgs/blob/bd4f00ac7fa6b641777b5a7fb08911c456b4ed01/pkgs/applications/editors/vim/plugins/overrides.nix#L197-L199)?

Should be done now, please recheck.

**Review by PerchunPak — APPROVED** [review]
LGTM

CC @GaetanLepage

File: pkgs/applications/editors/vim/plugins/overrides.nix [hunk 1]
```diff
@@ -561,6 +561,10 @@ in
     dependencies = with self; [ copilot-lua plenary-nvim ];
   };
 
+  copilot-lualine = super.copilot-lualine.overrideAttrs {
+    dependencies = with self; [ copilot-lua lualine-nvim ];
+  };
+
```
Comment (GaetanLepage) [inline pkgs/applications/editors/vim/plugins/overrides.nix:567]: Can we try to throw a
```
doInstallCheck ) true;
nvimRequireCheck = "copilot-lua";
```
here ?

Comment (rsniezek) [reply pkgs/applications/editors/vim/plugins/overrides.nix:567]: We can. the nvimRequireCheck should have only copilot-lua or also lualine-nvim? I'm not familiar with this part here

Comment (GaetanLepage) [reply pkgs/applications/editors/vim/plugins/overrides.nix:567]: It should be `copilot-lualine` sorry. No need to check the dependencies.

Comment (rsniezek) [reply pkgs/applications/editors/vim/plugins/overrides.nix]: please check

**Review by GaetanLepage — APPROVED** [review]

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 344278`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.copilot-lualine</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.copilot-lualine</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.copilot-lualine</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.copilot-lualine</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Pull request merged**